packages/gallery/src/components/gallery/leanGallery/leanGallery.js

```diff
@@ -40,6 +41,20 @@ export default class LeanGallery extends React.Component {
 
   componentDidMount() {
     this.eventsListener(GALLERY_CONSTS.events.APP_LOADED, {});
+    this.measureIfNeeded();
+    this.setState({
+      numberOfColumns: this.calcNumberOfColumns(this.props)
+    });
+  }
+
+  componentWillReceiveProps(nextProps) {
+    this.measureIfNeeded();
```

Comment:
this should be under the same condition:
`(this.props.container.galleryWidth !== nextProps.container.galleryWidth)`